aspnetcore/security/enforcing-ssl/sample-snapshot/3.x/Program.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace WebHTTPS
+{
+    #region snippet_Program
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.UseSetting("https_port", "8080")
```

Comment:
UseSetting is not a route we should encourage, it requires code changes. Show it in the appsettings.json instead.

Reply:
@scottaddie can you fix this? @tdykstra  has moved over to .NET

Reply:
cc @guardrex

Reply:
@Tratcher Are you requesting that we move the port number to *appsettings.json* and retrieve from there via the Configuration API? Or are you suggesting something different?

Reply:
If you define it in appsettings.json you don't need to retrieve it from config, that will happen automatically.

Reply:
It's actually *Properties\launchSettings.json*

```
{
  "iisSettings": {
    "windowsAuthentication": false,
    "anonymousAuthentication": true,
    "iisExpress": {
      "applicationUrl": "http://localhost:49637/",
      "sslPort": 0
    }
  },
  "profiles": {
    "IIS Express": {
      "commandName": "IISExpress",
      "launchBrowser": true,
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    },
    "ResponseFormattingSample": {
      "commandName": "Project",
      "launchBrowser": true,
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      },
      "applicationUrl": "http://localhost:49638/"
    }
  }
}
```

Reply:
Or can you define it in *appsetting.json* too?

Reply:
> Or can you define it in _appsetting.json_ too?

Yes, appsettings.json also works. And you can give different values in appsettings.development.json, etc..

Reply:
Merging this PR to master. I'll have a follow-up patch PR with the changes suggested by Chris.